Jellyfin.Server.Implementations/Activity/ActivityManager.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Jellyfin.Data.Entities;
+using MediaBrowser.Model.Activity;
+using MediaBrowser.Model.Events;
+using MediaBrowser.Model.Querying;
+
+namespace Jellyfin.Server.Implementations.Activity
+{
+    /// <summary>
+    /// Manages the storage and retrieval of <see cref="ActivityLog"/> instances.
+    /// </summary>
+    public class ActivityManager : IActivityManager
+    {
+        /// <inheritdoc/>
+        public event EventHandler<GenericEventArgs<ActivityLogEntry>> EntryCreated;
+
+        private JellyfinDbProvider _provider;
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="ActivityManager"/> class.
+        /// </summary>
+        /// <param name="provider">The Jellyfin database provider.</param>
+        public ActivityManager(JellyfinDbProvider provider)
+        {
+            _provider = provider;
+        }
+
+        /// <inheritdoc/>
+        public void Create(ActivityLog entry)
+        {
+            using var dbContext = _provider.GetConnection();
+            dbContext.ActivityLogs.Add(entry);
+            dbContext.SaveChanges();
+
+            EntryCreated?.Invoke(this, new GenericEventArgs<ActivityLogEntry>(ConvertToOldModel(entry)));
+        }
+
+        /// <inheritdoc/>
+        public async Task CreateAsync(ActivityLog entry)
+        {
+            using var dbContext = _provider.GetConnection();
+            await dbContext.ActivityLogs.AddAsync(entry);
+            await dbContext.SaveChangesAsync();
+
+            EntryCreated?.Invoke(this, new GenericEventArgs<ActivityLogEntry>(ConvertToOldModel(entry)));
+        }
+
+        /// <inheritdoc/>
+        public QueryResult<ActivityLogEntry> GetActivityLogEntries(DateTime? minDate, bool? hasUserId, int? startIndex, int? limit)
```

Comment:
IMO this is definitely a good change and not out of scope for this PR. In the initial conversion I tried to match as closely as possible with the original methods, but since I am rewriting the class anyways, changing method signatures a bit more is not a problem.